pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2226,6 +2226,53 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+
+	if IsBackupObjectTypeVirtualMachine(backup) {
```

Comment:
Suggestion: We can add a guard clause here instead of specifying the action when this condition is true. Something like this:
```
if !IsBackupObjectTypeVirtualMachine(backup) {
  return nil
 }
 preExecRule := backup.Spec.PreExecRule
postExecRule := backup.Spec.PostExecRule

deleteRuleIfExists := func(ruleName string) error {...
...
return nil
}